examples/aligned_accessor/aligned_accessor.cpp

```diff
@@ -0,0 +1,737 @@
+    // Passing in ba doesn't help the compiler
+    // deduce the template parameters,
+    // so we just specify them explicitly.
+    add_omp_aligned_simd_aligned_raw_1d<ElementType, byte_alignment>(n, x_blessed, y_blessed, z_blessed);
+  }
+  return TOCK();
+}
+#endif // _OPENMP
+
+template<class ElementType>
+auto warmup(const std::size_t num_trials,
+	    const index_type n,
+	    const ElementType x[],
+	    const ElementType y[],
+	    ElementType z[])
+{
+  return benchmark_add_raw_1d(num_trials, n, x, y, z);
+}
+
+} // namespace (anonymous)
+
+int main(int argc, char* argv[])
+{
+  using std::cout;
+  using std::cerr;
+  using std::endl;
+
+  if(argc != 3) {
+    cerr << "Usage: main <n> <num_trials>" << endl;
+    return -1;
+  }
+
+  int n = 10000;
+  int num_trials = 100;
+  n = std::stoi(argv[1]);
+  num_trials = std::stoi(argv[2]);
+
+  aligned_array<float, min_byte_alignment> x(n);
+  aligned_array<float, min_byte_alignment> y(n);
+  aligned_array<float, min_byte_alignment> z(n);
+
+  constexpr std::integral_constant<std::size_t, min_byte_alignment> byte_alignment;
+
+  auto warmup_result =
+    warmup(num_trials, n, x.data_unaligned(), y.data_unaligned(), z.data_unaligned());
+
+  auto aligned_mdspan_result =
+    benchmark_add_aligned_mdspan_1d(num_trials, n, x.data_aligned(),
+				    y.data_aligned(), z.data_aligned(),
+				    byte_alignment);
+  auto mdspan_result =
+    benchmark_add_mdspan_1d(num_trials, n, x.data_unaligned(),
+			    y.data_unaligned(), z.data_unaligned());
+  auto raw_result =
+    benchmark_add_raw_1d(num_trials, n, x.data_unaligned(),
+			 y.data_unaligned(), z.data_unaligned());
+  auto aligned_raw_result =
+    benchmark_add_aligned_raw_1d(num_trials, n, x.data_aligned(),
+				 y.data_aligned(), z.data_aligned(),
+				 byte_alignment);
+
+#ifdef _OPENMP
+  auto omp_simd_aligned_mdspan_result =
+    benchmark_add_omp_simd_aligned_mdspan_1d(num_trials, n, x.data_aligned(),
+					     y.data_aligned(), z.data_aligned(),
+					     byte_alignment);
+  auto omp_simd_mdspan_result =
+    benchmark_add_omp_simd_mdspan_1d(num_trials, n, x.data_unaligned(),
+				     y.data_unaligned(), z.data_unaligned());
+  auto omp_simd_raw_result =
+    benchmark_add_omp_simd_raw_1d(num_trials, n, x.data_unaligned(),
+				  y.data_unaligned(), z.data_unaligned());
+  auto omp_aligned_simd_raw_result =
+    benchmark_add_omp_aligned_simd_raw_1d(num_trials, n,
+					  x.data_aligned(), y.data_aligned(), z.data_aligned(),
+					  byte_alignment);
+  auto omp_simd_aligned_raw_result =
+    benchmark_add_omp_simd_aligned_raw_1d(num_trials, n,
+					  x.data_aligned(), y.data_aligned(), z.data_aligned(),
+					  byte_alignment);
+  auto omp_aligned_simd_aligned_raw_result =
+    benchmark_add_omp_aligned_simd_aligned_raw_1d(num_trials, n,
+						  x.data_aligned(), y.data_aligned(), z.data_aligned(),
+						  byte_alignment);
+
+#endif // _OPENMP
+
+  cout << "Number of trials: " << num_trials << endl
```

Comment:
Flushing the stream excessively

Reply:
I'm not too worried about this, since this is the end of the benchmark and not performance-critical code.